store/snap_remote_repo.go

```diff
@@ -375,3 +401,55 @@ var download = func(name string, w io.Writer, req *http.Request, pbar progress.M
 
 	return err
 }
+
+type assertionSvcError struct {
+	Status int    `json:"status"`
+	Type   string `json:"type"`
+	Title  string `json:"title"`
+	Detail string `json:"detail"`
+}
+
+// Assertion retrivies the assertion for the given type and primary key.
+func (s *SnapUbuntuStoreRepository) Assertion(assertType *asserts.AssertionType, primaryKey ...string) (asserts.Assertion, error) {
+	url, err := s.assertionsURI.Parse(path.Join(assertType.Name, path.Join(primaryKey...)))
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest("GET", url.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	ssoToken, err := ReadStoreToken()
+	if err != nil {
+		s.setAuthHeader(req, ssoToken)
+	}
+	req.Header.Set("Accept", asserts.MediaType)
```

Comment:
@emgee kind of, we would set a lot more headers that aren't relevant to the assertion service